galasa-ui/src/app/mysettings/page.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+'use client'
+
+import { useEffect, useState } from "react";
+import { Loading, Button} from "@carbon/react"
+import PageTile from "@/components/PageTile";
+import styles from "../../styles/MySettings.module.css"
+import TokenCard from "@/components/TokenCard";
+import ErrorPage from "../error/page";
+import TokenRequestModal from "@/components/TokenRequestModal";
+import TokenDeleteModal from "@/components/TokenDeleteModal";
+import TokenResponseModal from "@/components/TokenResponseModal";
+
+export default function MySettingsPage() {
+
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
+    const [selectedTokens, setSelectedTokens] = useState<Set<string>>(new Set());
+    const [tokens, setTokens] = useState<Set<Token>>(new Set())
+    const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false)
+
+    interface Token {
+        tokenId: string;
+        description: string;
+        creationTime: string;
+        owner: {
+            loginId: string;
+        };
+    }
+
+    const handleSelectTokensForDeletion = (tokenId: any) => {
```

Comment:
Also, it's not clear what this is exactly doing since it's passed into the TokenCard component and called when you click on a token, acting like a toggle